python310Packages.hydra: 1.1.1 -> 1.2.0, python310Packages.omegaconf: 2.1.1 -> 2.2.2

onny opened the pull request:
###### Description of changes

update package and fix build (https://hydra.nixos.org/build/183008383)

also updating required dependencies

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

onny committed:
python310Packages.hydra: 1.1.1 -> 1.2.0

onny committed:
python310Packages.antlr4_9-python3-runtime: init at 4.9

onny committed:
python310Packages.omegaconf: 2.1.1 -> 2.2.2

bcdarwin commented:
Thanks!

According to `nixpkgs-review pr` this broke `python3-explorerscript` which now needs to be given `antlr4_8`.

onny commented:
> Thanks!
> 
> According to `nixpkgs-review pr` this broke `python3-explorerscript` which now needs to be given `antlr4_8`.

Can I handle this in a separate PR?

bcdarwin commented:
Probably individual PRs should be "logically consistent", or at least not introduce breakage, especially since in this case it's just a question of pushing another commit to bump `antlr` (or use `antlr4_9` instead).

onny commented:
> Probably individual PRs should be "logically consistent", or at least not introduce breakage, especially since in this case it's just a question of pushing another commit to bump `antlr` (or use `antlr4_9` instead).

I added another python runtime for antlr4_9 and used this in the two other python modules instead of bumping the general antlr4 version

bcdarwin commented on pkgs/development/python-modules/antlr4_9-python3-runtime/default.nix:
```diff
@@ -0,0 +1,23 @@
+buildPythonPackage rec {
+  pname = "antlr4_9-python3-runtime";
+  inherit (antlr4_9.runtime.cpp) version src;
+  disabled = !isPy3k;
```
Should probably be changed to `pythonOlder "3.5"` since it won't build without `typing` package otherwise (although surely no-one would notice this difference).

bcdarwin commented on pkgs/development/python-modules/antlr4_9-python3-runtime/default.nix:
```diff
@@ -0,0 +1,23 @@
+
+  # in 4.9, test was renamed to tests
+  checkPhase = ''
+    cd test*
```
Since we have a separate expression for 4.8 and 4.9, just `test` would be better here.

bcdarwin commented on pkgs/development/python-modules/antlr4_9-python3-runtime/default.nix:
```diff
@@ -0,0 +1,23 @@
+{ lib, buildPythonPackage, isPy3k, python
```
[Totally nitpicking here, but the single linebreak here looks a bit weird.  Consider running `nixpkgs-fmt` though.]

bcdarwin commented on pkgs/top-level/python-packages.nix:
```diff
@@ -563,6 +563,10 @@ in {
     inherit (pkgs) antlr4;
   };
 
+  antlr4_9-python3-runtime = callPackage ../development/python-modules/antlr4_9-python3-runtime {
```
Probably not too important but I feel the general style is something like `antlr4_8-python3-runtime = ...; antlr4_9-python3-runtime = ...; antlr4-python3-runtime = antlr4_9-python3-runtime` (preferably the latest version).

onny commented:
@bcdarwin Thank you for reviewing. I integrated your suggestions but also refactored the python antlr runtime packages a bit. I don't want to bump the antlr4 or antlr4 python runtime versions generally to avoid breakage of other packages so I just added version 4.8 for the python runtime

mweinelt commented:
We don't support multiple versions of the same library with python3Packages, because they will conflict when used, even if transitively.

SuperSandro2000 commented on pkgs/development/python-modules/antlr4-python3-runtime/default.nix:
```diff
@@ -1,11 +1,13 @@
   pname = "antlr4-python3-runtime";
   inherit (antlr4.runtime.cpp) version src;
-  disabled = !isPy3k;
+  disabled = python.pythonOlder "3.6";
```
```suggestion
  disabled = pythonOlder "3.6";
```
